app/src/main/java/com/example/stromtracker/ui/geraete/GeraeteFragment.kt

```diff
@@ -81,101 +83,103 @@ class GeraeteFragment : Fragment(), View.OnClickListener {
         geraeteList = ArrayList()
         produzentList = ArrayList()
         kategorieList = ArrayList()
-        raumList = ArrayList()
-
-        val navView = requireActivity().findViewById<NavigationView>(R.id.nav_view)
+        raumListHaushalt = ArrayList()
 
-        val sp: Spinner = navView.menu.findItem(R.id.nav_haushalt).actionView as Spinner
-        currHaushalt = sp.selectedItem as Haushalt
+        var mainAct = requireActivity() as MainActivity
+        iconArray = mainAct.getIconArray()
 
 
 
         viewManager = LinearLayoutManager(this.context)
-        viewAdapter = GeraeteListAdapter(geraeteList, kategorieList, raumList)
+        viewAdapter = GeraeteListAdapter(geraeteList, kategorieList, raumListHaushalt, iconArray)
         recyclerView = root.findViewById<RecyclerView>(R.id.geraete_recycler_view).apply {
             setHasFixedSize(true)
             layoutManager = viewManager
             adapter = viewAdapter
         }
 
         produzentViewManager = LinearLayoutManager(this.context)
-        produzentViewAdapter = GeraeteListAdapter(produzentList, kategorieList, raumList)
-        produzentRecyclerView = root.findViewById<RecyclerView>(R.id.produzent_recycler_view).apply {
-            setHasFixedSize(true)
-            layoutManager = produzentViewManager
-            adapter = produzentViewAdapter
-        }
+        produzentViewAdapter =
+            GeraeteListAdapter(produzentList, kategorieList, raumListHaushalt, iconArray)
+        produzentRecyclerView =
+            root.findViewById<RecyclerView>(R.id.produzent_recycler_view).apply {
+                setHasFixedSize(true)
+                layoutManager = produzentViewManager
+                adapter = produzentViewAdapter
+            }
         return root
     }
 
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
 
-
-
         geraeteViewModel = ViewModelProviders.of(this).get(GeraeteViewModel::class.java)
+        sharedViewModel = ViewModelProviders.of(requireActivity()).get(SharedViewModel::class.java)
```

Comment:
Jetzt hast du 2 View Models?

Reply:
Ja, eins vom GeräteFragment und eins nur für den Haushalt.

Aber wie oben geschrieben, wenn wir alle ViewModel in eins mergen dann hat sich das eh erledigt.

Reply:
ok ja da müssen wir dann nochmal alle überlegen